### DISCUSSION TITLE
Curious about current status of this project ?

### DISCUSSION THREAD

**Discussion by ghost** [navarrothiago/upf-bpf#69] [General]
Hi @navarrothiago, I love this project but lately I dont see much happening. What is the status of this project ?

**Comment by navarrothiago** [comment]
Thanks, @PacketCrunch. It's good to know about that. Unfortunately, I'm not maintaining the project anymore. However, if you're interested in contributing, please let me know. I'd be glad to help with that.

**Reply by ghost** [reply]
Oh I see @navarrothiago. I would love to do something. Its a shame to see a good project gather dust. One quick question - which SMF did you integrate the UPF with ??

**Reply by navarrothiago** [reply]
Oh, sorry for that. Just to put some context.. The project was initiated during my MSc program, where I devoted the majority of my time to it. After my defense, I was assigned to different activities. The project's aim is to develop a UPF-BPF core library that can enable third-party UPF to utilize BPF technology. The first step is to integrate the UPF-BPF library into SPGWU/UPF (OAI), as seen in this discussion (https://github.com/navarrothiago/upf-bpf/discussions/52). The UPF-BPF doesn't implement the PFCP protocol, as it was merely a proof of concept to validate the idea. However, with additional effort, it could potentially be expanded to support more functionalities. In response to your question, we haven't integrated with any SMF. I have been in communication with the OAI team who are attempting to integrate with their UPF component, but I'm currently unaware of its status.

**Reply by ghost** [reply]
Thanks a lot for the explanation !!
